src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java

```diff
@@ -0,0 +1,95 @@
+package de.tum.in.www1.artemis.service;
+
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.repository.StatisticsRepository;
+
+@Service
+public class StatisticsService {
+
+    private final StatisticsRepository statisticsRepository;
+
+    public StatisticsService(StatisticsRepository statisticsRepository) {
+        this.statisticsRepository = statisticsRepository;
+    }
+
+    public Integer getLoggedInUsers(Long span) {
+        return this.statisticsRepository.getLoggedInUsers(ZonedDateTime.now().minusDays(span).toInstant());
+    }
+
+    public Integer getActiveUsers(Long span) {
+        return this.statisticsRepository.getActiveUsers(ZonedDateTime.now().minusDays(span));
+    }
+
+    /**
+     * Forwards the request to the repository, which returns a List<Map<String, Object>>, with String being the column name, "day" and "amount" and Object being the value,
+     * either the date or the amount of submissions. It then collects the amounts in an array, depending on the span value, and returns it
+     *
+     * @param span DAY,WEEK,MONTH or YEAR depending on the active tab in the view
+     * @return a array, containing the values for each bar in the graph
+     */
+    public Integer[] getTotalSubmissions(String span) {
+        switch (span) {
+            case "DAY": // result = this.statisticsRepository.getTotalSubmissionsDay(ZonedDateTime.now().minusDays(7));
+                return null;
+            case "WEEK":
+                Integer[] result = new Integer[7];
+                Arrays.fill(result, 0);
+                ZonedDateTime border = ZonedDateTime.now().minusDays(6).withHour(0).withMinute(0).withSecond(0);
```

Comment:
This does not what I would expect. If I click on the week tab i would expect it to show me the values for THIS week and not just go back 7 days, same for month and so on. I mean if it currently wednesday I expect it to show me the values for monday, tuesday and wednesday. Does anybody else have an opinion here? Or you have to rename the client to something like `Last 7 Days` `Last 30 Days`

Reply:
Yes this is a debatable issue. However, I orientate myself roughly on the Apple Health App, which also displays the last 7/30 etc. days. Plus it displays more information, especially if its Monday and there is no data yet, this seems a bit weird IMO